### PR TITLE
issue1549 durack1 update Matplotlib 1.4.3 to 1.5.0

### LINKED ISSUE #1549: Update matplotlib to 1.5.0
The [`matplotlib 1.5.0RC1` just landed](http://matplotlib.org/devdocs/users/whats_new.html#new-in-matplotlib-1-5), so should be considered for upgrading when the full release is out - likely on a `UV-CDAT 3.0` timeline

### PR DESCRIPTION
**Pull request by durack1**
Fix #1549 - Update Matplotlib 1.4.3 to 1.5.0

### CODE REVIEW

**Comment by doutriaux1** [comment]
failed on RH6

**Comment by durack1** [comment]
@doutriaux1 looks like the `matplotlib` failure [isn't the only one](https://open.cdash.org/viewTest.php?onlyfailed&buildid=4079644)..

```
The following tests FAILED:
	 88 - vcs_test_background_mode_rotate (Failed)
	515 - matplotlib_verify_import (Failed)
	516 - basemap_verify_import (Failed)
```

With `matplotlib` it seems like a [package `cycler` isn't installed](https://open.cdash.org/testDetails.php?test=386576218&build=4079644):
```
Successfully updated your environment to use UVCDAT
(changes are valid for this session/terminal only)
Version: 2.4.0.rc2-36-g1ed6144
Location: /export/doutriaux1/uvcbot/build/install
Traceback (most recent call last):
  File "/export/doutriaux1/uvcbot/uvcdat/testing/matplotlib/test_matplotlib_import.py", line 2, in <module>
    import matplotlib
  File "/export/doutriaux1/uvcbot/build/install/lib/python2.7/site-packages/matplotlib-1.5.0-py2.7-linux-x86_64.egg/matplotlib/__init__.py", line 124, in <module>
    from matplotlib.rcsetup import (defaultParams,
  File "/export/doutriaux1/uvcbot/build/install/lib/python2.7/site-packages/matplotlib-1.5.0-py2.7-linux-x86_64.egg/matplotlib/rcsetup.py", line 29, in <module>
    from cycler import Cycler, cycler as ccycler
ImportError: No module named cycler
```
I'll take another look at the `matplotlib` dependency list..

**Comment by durack1** [comment]
@doutriaux1 yep, looks like [`cycler` is a required package](http://matplotlib.org/users/installing.html#required-dependencies) I'll add it in..

Are the other dependencies also available: `python, numpy (1.6+), dateutil (1.1+), pyparsing, libpng (1.2+), pytz, freetype (2.3+), cycler (0.9+)`

It also ships with: 'agg 2.4, qhull 2012.1, ttconv, six 1.9.0` so there might be some duplication with existing source being built here.. The build order might matter..

File: CMake/cdat_modules/cycler_pkg.cmake [hunk 1]
```diff
@@ -0,0 +1,10 @@
+set(CYCLER_MAJOR 0)
+set(CYCLER_MINOR 9)
+set(CYCLER_PATCH 0)
+set(CYCLER_VERSION ${CYCLER_MAJOR}.${CYCLER_MINOR}.${CYCLER_PATCH})
+set(CYCLER_URL ${LLNL_URL})
+set(CYCLER_GZ cycler-${CYCLER_VERSION}.tar.gz)
+set(CYCLER_MD5 c10ade5ca3f0aadf575eb25203b225a5)
+set(CYCLER_SOURCE ${CYCLER_URL}/${CYCLER_GZ})
+
+add_cdat_package_dependent(Cycler "" "" OFF "CDAT_BUILD_LEAN" OFF)
```
Comment (durack1) [inline CMake/cdat_modules/cycler_pkg.cmake]: @doutriaux1 not certain that this magic is correct - I used the template from `Cython`

**Comment by durack1** [comment]
@doutriaux1 do you know any reason why [`crunchy` failed](http://aims4.llnl.gov:9981/master/crunchy.llnl.gov:9982/UV-CDAT/uvcdat/5dbf38b3fc6a52b0c6b635cdd92dd00ae8e1db02/ctest__-j12__-D), yet [`oceanonly` passed](http://aims4.llnl.gov:9981/master/oceanonly.llnl.gov:9982/UV-CDAT/uvcdat/5dbf38b3fc6a52b0c6b635cdd92dd00ae8e1db02/ctest__-j12__-D)?

**Comment by durack1** [comment]
@doutriaux1 this is weird, why is [`crunchy` failing](https://open.cdash.org/testDetails.php?test=387924390&build=4085484) - it seems to be out of sync with the PR..

It seems there's a [`pathspec` error](http://aims4.llnl.gov:9981/master/crunchy.llnl.gov:9982/UV-CDAT/uvcdat/2b369a8cbf1d1f53bd4db65e246e0ee33ac8c4de/git__checkout__issue1549_durack1_UpdateMatplotlib1.4.3To1.5.0) (`error: pathspec 'issue1549_durack1_UpdateMatplotlib1.4.3To1.5.0' did not match any file(s) known to git.`) which indeed is causing grief on `crunchy`

**Comment by doutriaux1** [comment]
no this error is `normal` it tries to find a matching baseline branch and it's an acceptable and expected failure in this case

**Comment by durack1** [comment]
@doutriaux1 any idea what's happening with [`Darwin-Mac2`](http://aims4.llnl.gov:9981/master/macbot2.llnl.gov:9982/UV-CDAT/uvcdat/68da20927442b220c4f8f8d33869055b2617a377/make__-j4) it seems `Cycler` and `Matplotlib` installed fine, just the [import went awry](https://open.cdash.org/testDetails.php?test=387948178&build=4085638)..

```
Successfully updated your environment to use UVCDAT
(changes are valid for this session/terminal only)
Version: 2.4.0.rc2-44-g68da209
Location: /Users/doutriaux1/bot/build/install
Traceback (most recent call last):
  File "/Users/doutriaux1/bot/uvcdat/testing/matplotlib/test_matplotlib_import.py", line 2, in <module>
    import matplotlib
ImportError: No module named matplotlib
```

On [`crunchy` seems like the normal issues were reported](http://aims4.llnl.gov:9981/master/crunchy.llnl.gov:9982/UV-CDAT/uvcdat/68da20927442b220c4f8f8d33869055b2617a377/ctest__-j12__-D):
```
The following tests FAILED:
	 88 - vcs_test_background_mode_rotate (Failed)
	435 - vcs_addons_test_EzTemplate_12_plots_mix_glb_local (Failed)
```

**Comment by durack1** [comment]
@doutriaux1 wanna take a peek at this - I also think this PR is *almost* ready to go..

**Comment by durack1** [comment]
@jbeezley any idea what is going on with `garant` it seems it's failing a completely different set of tests..?

**Comment by jbeezley** [comment]
It looks like there are some data files it isn't getting.  I've seen this before, but I have no idea where the `diags_test` data comes from.

**Comment by jbeezley** [comment]
The `diags_test`'s were broken by #1615.  The files are never downloaded and the directory that is supposed to contain them `build/install/share` does not exist.

**Comment by durack1** [comment]
@jbeezley I wonder why only `garant` was reporting this failure?

**Comment by jbeezley** [comment]
Good question.  I see this in the [oceanonly build](http://aims4.llnl.gov:9981/master/oceanonly.llnl.gov:9982/UV-CDAT/uvcdat/68da20927442b220c4f8f8d33869055b2617a377/cmake___export_doutriaux1_bots_oceanonly_uvcdat__-DCDAT_CHECKOUT_BASELINE_MATCHING_BRANCH=OFF):
```
[INFO] Failed to fetch test data for uvcmetrics, tests will fail
```

It is a mystery how the tests are passing.

**Comment by jbeezley** [comment]
It is executing a python script using the uvcdat python before it is built.  Again, I have no idea why the LLNL builds are working unless the build directory is not getting deleted between builds.

**Comment by doutriaux1** [comment]
The build directories are being systematically deleted.

**Comment by durack1** [comment]
@doutriaux1 I think this is good to go

**Comment by durack1** [comment]
@doutriaux1 I think this is good to go

**Pull request merged**